### DISCUSSION TITLE
Button field validation

### DISCUSSION THREAD

**Discussion by SergeyWebDeveloper** [react-hook-form/react-hook-form#10212] [Q&A]
Good afternoon. I have 2 buttons. When I click on one of them, I have to check all the form fields, and when I click on the other, only a small part of certain fields should be checked. I can't figure out how to achieve this. You can offer? 

sketched a small example to help
https://codesandbox.io/s/register-forked-szquch?file=/src/App.js

**Comment by SergeyWebDeveloper** [comment]
@bluebill1049 , any ideas pls?)

**Comment by leapful** [comment]
@SergeyWebDeveloper 

You can perform `trigger()` on fields you want to validate via `onClick` handler of the button.

**Comment by SergeyWebDeveloper** [comment]
@leapful , the fact is that some fields are optional, but you need to check the min and max characters. I don't understand how to check this separately.
